src/util/Profiler.h

```diff
@@ -0,0 +1,57 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2022, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <chrono>
+#include <type_traits>
+#include <utility>
+
+namespace util {
+
+/// @brief Profiler function to measure the time consuming
+/// @param func function object, can be a lamdba or function wrapper
+/// @return return a pair if function wrapper has return value: result of
+/// function wrapper and the elapsed time(ms) during executing the given
+/// function only return the elapsed time if function wrapper does not have
+/// return value
+template <typename F>
+auto
```

Comment:
Let's add `[[nodiscard]]` here

Reply:
And `noexcept` but only if we can guarantee `duration_cast` will not throw in this situation. Might not be feasible if we allow customization thru tparam `DurationType` tho.

Reply:
[nodiscard] added. I did not add "noexcept", because if the func throws exception, the whole process will be terminated.